src/components/Transactions.js

```diff
@@ -27,15 +32,28 @@ class Transactions extends React.Component {
       loaded: false,
       hasAfter: false,
       hasBefore: false,
+      queryParams: this.obtainQueryParams(),
     }
   }
 
   componentDidMount() {
-    this.getData(true, null, null, '');
+    this.getData(this.state.queryParams);
 
     WebSocketHandler.on('network', this.handleWebsocket);
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    const queryParams = this.obtainQueryParams();
+
+    // Do we have new URL params?
+    if (JSON.stringify(this.state.queryParams) !== JSON.stringify(queryParams)) {
```

Comment:
You could use the mehtod `shouldComponentUpdate` to avoid rerendering if the url didn't change. 

> Use `shouldComponentUpdate()` to let React know if a component’s output is not affected by the current change in state or props.
> 
> https://reactjs.org/docs/react-component.html#shouldcomponentupdate

Reply:
If we decide to do it we should take care because the URL change is not the only thing that have to trigger a render on the page. One other thing that might do it is a ws message with a new transaction for the page (we re-render it in case we are in the first page).

I am still not 100% sure if it's worth it to do it now, since the page has a good performance. Following react's docs:

> This method only exists as a performance optimization. Do not rely on it to “prevent” a rendering, as this can lead to bugs.

Of course it would be good but we just need to take care and think about all possible scenarios.

Reply:
Thanks for the comments, guys. I plan to leave it as is since it was never meant to prevent rendering. Although it could, I agree with pedroferreira1 that it shouldn't. This logic only decides whether or not to trigger a new backend fetch call by comparing the current URL params with the params of the last processed backend response.